jbpm-designer-backend/src/main/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnit.java

```diff
@@ -618,9 +644,15 @@ public void evaluateWorkDefinitions(Map<String, WorkDefinitionImpl> workDefiniti
 
                 Map<String, Object> parameterValues = new HashMap<>();
                 if (workDefinitionMap.get("parameterValues") != null) {
+                    ClassLoader parentLoader = Thread.currentThread().getContextClassLoader();
                     try {
                         Map<String, Object> parameterValuesMap = (Map<String, Object>) workDefinitionMap.get("parameterValues");
                         if (parameterValuesMap != null) {
+                            ClassLoader contextLoader = null;
```

Comment:
I can refactor the code to only do the work associated with creating a new class loader when there is an EnumDataType.